src/diagrams/common/common.ts

```diff
@@ -0,0 +1,147 @@
+import DOMPurify from 'dompurify';
+import { MermaidConfig } from 'types/config';
+
+/**
+ * Gets the rows of lines in a string
+ *
+ * @param {string | undefined} s The string to check the lines for
+ * @returns {string[]} The rows in that string
+ */
+export const getRows = (s?: string): string[] => {
+  if (!s) return [''];
+  const str = breakToPlaceholder(s).replace(/\\n/g, '#br#');
+  return str.split('#br#');
+};
+
+/**
+ * Removes script tags from a text
+ *
+ * @param {string} txt The text to sanitize
+ * @returns {string} The safer text
+ */
+export const removeScript = (txt: string): string => {
+  return DOMPurify.sanitize(txt);
+};
+
+const sanitizeMore = (text: string, config: MermaidConfig) => {
+  // TODO Q: Should this check really be here? Feels like we should be sanitizing it regardless.
+  if (config.flowchart?.htmlLabels !== false) {
```

Comment:
`htmlLabels` should be moved up from flowchart to config.
Will make this a separate issue and handle it there.

Reply:
Yes, we have several of those. When we change them we need to keep track of them perhaps we can make compatibility mode for easy upgrade.

Reply:
We can skip the check. But a warning, sometimes we get too rigid like brs in tool tips showing as the tag instead of a new line.